### ISSUE TITLE
Why are we using nn.dropout instead of layers.droput in the ptb example ?

### ISSUE THREAD

**Issue by grzegorznowak** [tensorflow/models#3181]
https://github.com/tensorflow/models/blob/9325ea8feb5458d0ba2046a0134c305fd92e82a5/tutorials/rnn/ptb/ptb_word_lm.py#L132

Is it right to use `nn.dropout` here instead of `layers.droput` ? If so why do we still want to block neurons randomly while validating/testing the model ? Can we add a comment to the code/doc please to explain ?

`I'm still an uber noob to graphs/TF, so please pardon my ignorance if I'm asking pure rubbish.`

**Comment by aselle** [comment]
Likely, `tf.layers.dropout` didn't exist when the original model was written. `tf.layers.dropout` is nicer in that it has a is_training entry, so you don't need to conditionally create different graphs for inference/training. If you would like to submit a comment, or try  to replace the tf.nn.dropout with tf.layers.dropout, your contribution would be welcome.

**Comment by grzegorznowak** [comment]
ah right thanks @aselle , yes I can see how we create that conditionally, depending on the state of a model. Yes would love to post PR for gaining more confidence into the matter, but not until my brain switches to thinking in the Graph. cheers!

**Comment by grzegorznowak** [comment]
sorry, not sure if it was ok I closed the issue, so feel free to reopen if it's better to float as a thing-to-do of course.
